closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -343,40 +357,48 @@ int encrypt(byte[] in, int inOfs, int len) {
      */
     int encryptFinal(byte[] in, int inOfs, int len, byte[] out, int outOfs)
         throws IllegalBlockSizeException, ShortBufferException {
+        byte[] keyCopy;
+        byte[] ivCopy;
+        byte[] aad;
+        int tagLenBytes = this.tagLenBytes;
+        synchronized (this) {
+            if (len > MAX_BUF_SIZE - tagLenBytes) {
+                throw new ShortBufferException
+                    ("Can't fit both data and tag into one buffer");
+            }
 
-        if (len > MAX_BUF_SIZE - tagLenBytes) {
-            throw new ShortBufferException
-                ("Can't fit both data and tag into one buffer");
-        }
+            if (out.length - outOfs < (len + tagLenBytes)) {
+                throw new ShortBufferException("Output buffer too small");
+            }
 
-        if (out.length - outOfs < (len + tagLenBytes)) {
-            throw new ShortBufferException("Output buffer too small");
-        }
+            checkDataLength(ibuffer_enc.size(), len);
 
-        checkDataLength(ibuffer_enc.size(), len);
+            if (len > 0) {
+                ibuffer_enc.write(in, inOfs, len);
+            }
 
-        if (len > 0) {
-            ibuffer_enc.write(in, inOfs, len);
-        }
+            // refresh 'in' to all buffered-up bytes
+            in = ibuffer_enc.toByteArray();
+            ibuffer_enc.reset();
 
-        // refresh 'in' to all buffered-up bytes
-        in = ibuffer_enc.toByteArray();
-        inOfs = 0;
-        len = in.length;
-        ibuffer_enc.reset();
+            aad = ((aadBuffer == null || aadBuffer.size() == 0) ? emptyAAD : aadBuffer.toByteArray());
```

Comment:
Missing parentheses around `==` expressions.

Reply:
Added paranthesis around `aadBuffer == null` and around `aadBuffer.size() == 0`.